test/fixtures/azure-fixture/src/main/java/fixture/azure/MockAzureBlobStore.java

```diff
@@ -0,0 +1,451 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+package fixture.azure;
+
+import org.elasticsearch.ExceptionsHelper;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.bytes.CompositeBytesReference;
+import org.elasticsearch.common.util.Maps;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.rest.RestStatus;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public class MockAzureBlobStore {
+
+    private static final Logger logger = LogManager.getLogger(MockAzureBlobStore.class);
+    private static final String BLOCK_BLOB_TYPE = "BlockBlob";
+    private static final String PAGE_BLOB_TYPE = "PageBlob";
+    private static final String APPEND_BLOB_TYPE = "AppendBlob";
+
+    private final LeaseExpiryPredicate leaseExpiryPredicate;
+    private final Map<String, AzureBlockBlob> blobs;
+
+    /**
+     * Provide the means of triggering lease expiration
+     *
+     * @param leaseExpiryPredicate A Predicate that takes an active lease ID and returns true when it should be expired, or null to never fail leases
+     */
+    public MockAzureBlobStore(@Nullable LeaseExpiryPredicate leaseExpiryPredicate) {
+        this.blobs = new ConcurrentHashMap<>();
+        this.leaseExpiryPredicate = leaseExpiryPredicate != null ? leaseExpiryPredicate : (current, request) -> false;
```

Comment:
Could we introduce a constant `LeaseExpiryPredicate#NEVER` or similar rather than using `null` for that case? I think it'd be more readable in the callers.

Reply:
Yep, addressed in 453283b033d2e09b05efeff41501a13d6ef2739c